### ISSUE TITLE
3.0.0b4 HTTPS proxy support

### ISSUE THREAD

**Issue by andreybavt** [aio-libs/aiohttp#2722]
Hi,

Do you support https proxies? Documentation says that yes, but if I try to pass one to post method it throws 
`Only http proxies are supported`.

I'm using version 3.0.0b4 
```python
async with aiohttp.ClientSession() as session:
    async with session.post(url, proxy='https://192.116.142.153:8080',
                            headers={"X-Requested-With": "XMLHttpRequest"}, timeout=5) as response:
```

Thanks

**Comment by XayOn** [comment]
I think this issue would be of interest: https://github.com/aio-libs/aiohttp/issues/845#issuecomment-278727807

**Comment by asvetlov** [comment]
The problem is:
- We support plain HTTP proxies.
- We support HTTPS proxies with connection to proxy server by plain HTTP and upgrading to HTTPS by `CONNECT` HTTP method: https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/CONNECT
- Honestly I have no idea what HTTPS proxy is. What is the protocol to communicate to?

Maybe somebody can shed a light?

**Comment by asvetlov** [comment]
As of Python 3.7 asyncio itself doesn't support HTTPS proxy mode (starting TLS in already TLS channel).
3.8 has a chance to fix it.
Nothing to do on aiohttp side now, sorry.

**Comment by cooperlees** [comment]
Can we open this back up OR should I create a new issue to test + make this work once https://bugs.python.org/issue34971 is closed with 3.8?
- It seems there is a PR up (https://github.com/python/cpython/pull/9840) but it needs a redesign? Have prodded.

**Comment by asvetlov** [comment]
No, sorry. 
https://bugs.python.org/issue34971 is for TLS tickets which are not related.

What we need for HTTPS proxies is support `loop.start_tls()` when called from an already encrypted connection (TLS in TLS).
AFAIK `uvloop` supports it already (@1st1 told me that it does in recent versions)  but sslproto implementation used by uvloop is not backported to asyncio yet.

Sorry, I had no time to do it personally, was busy on delivering other asyncio features.

FYI aiohttp doesn't utilize Python 3.7 `loop.start_tls()` (implemented by Yuri) and `loop.sendfile()` (my contribution) yet but uses custom workarounds because we need to support Python 3.5.3+ now.

We have to switch to the officially supported approach eventually. If `loop.start_tls()` start to provide TLS in TLS would be a strong motivation.
